lib/text/vtt_text_parser.js

```diff
@@ -219,6 +222,75 @@ shaka.text.VttTextParser = class {
     return cue;
   }
 
+  /**
+   * Parses a WebVTT styles from the given payload.
+   *
+   * @param {string} payload
+   * @param {!shaka.text.Cue} rootCue
+   */
+  static parseCueStyles(payload, rootCue) {
+    const xmlPayload = '<span>' + payload + '</span>';
+    const element = shaka.util.XmlUtils.parseXmlString(xmlPayload, 'span');
+    if (element) {
+      /** @type {Array.<!shaka.extern.Cue>} */
+      const cues = [];
+      const VttTextParser = shaka.text.VttTextParser;
+      const childNodes = [...element.childNodes];
+      if (childNodes.length == 1) {
+        const childNode = childNodes[0];
+        if (childNode.nodeType == Node.TEXT_NODE ||
+            childNode.nodeType == Node.CDATA_SECTION_NODE) {
+          rootCue.payload = payload;
+          return;
+        }
+      }
+      for (const childElement of childNodes) {
+        VttTextParser.generateCueFromElement_(childElement, rootCue, cues);
+      }
+      rootCue.nestedCues = cues;
```

Comment:
You're generating a flat list of nestedCues, which is fine and should work.  But we just landed changes to the text displayers to support deeply-nested cues, and the TTML parser will soon start generating such cues.  If you like, you can do that here, too.

Reply:
I prefer to leave this as is now since it is working. If for you it is a problem that this change cannot be merged, I will change it.

Reply:
No, it's fine as is.  Just wanted to inform you of options that you didn't have before you started.